diff --git a/tm2/pkg/amino/deep_copy.go b/tm2/pkg/amino/deep_copy.go
index 669722be4..a8a186fcd 100644
--- a/tm2/pkg/amino/deep_copy.go
+++ b/tm2/pkg/amino/deep_copy.go
@@ -51,6 +51,9 @@ func deepCopy(src, dst reflect.Value) {
 }
 
 func _deepCopy(src, dst reflect.Value) {
+	if isNil(src) {
+		return
+	}
 	switch src.Kind() {
 	case reflect.Pointer:
 		cpy := reflect.New(src.Type().Elem())
@@ -84,10 +87,6 @@ func _deepCopy(src, dst reflect.Value) {
 		}
 
 	case reflect.Slice:
-		if src.IsNil() {
-			dst.Set(src)
-			return
-		}
 		switch src.Type().Elem().Kind() {
 		case reflect.Int64, reflect.Int32, reflect.Int16,
 			reflect.Int8, reflect.Int, reflect.Uint64,
